contracts/Staking.sol

```diff
@@ -386,11 +384,29 @@ contract Staking is IStaking, Initializable, OwnableUpgradeable, Constants {
         uint256 index = unbondingLength[_source];
         unbondingAmount[_source][index].amount = _amount;
         unbondingAmount[_source][index].startTime = block.timestamp;
+        unbondingAmount[_source][index].indexer = _indexer;
         unbondingLength[_source]++;
 
         emit UnbondRequested(_source, _indexer, _amount, index);
     }
 
+    function cancelUnbonding(uint256 unbondReqId) external {
+        require(unbondReqId >= withdrawnLength[msg.sender], 'already withdrawed');
+        UnbondAmount memory unbond = unbondingAmount[msg.sender][unbondReqId];
+        IIndexerRegistry indexerRegistry = IIndexerRegistry(settings.getIndexerRegistry());
+        require(indexerRegistry.isIndexer(unbond.indexer), 'indexer unregistered');
+
+        delete unbondingAmount[msg.sender][unbondReqId];
+        if (msg.sender != unbond.indexer) {
+            require(
+                delegation[unbond.indexer][unbond.indexer].valueAfter * indexerLeverageLimit >=
```

Comment:
Same logic with the one in `functiondelegate`, can extract to a separate function: `checkDelegateLimitation(address _indexer, uint256 _amount)`